src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,656 @@
+            {
+                if (!parameterSymbol.HasExplicitDefaultValue && !parameterSymbol.Type.Equals(compilation.GetTypeByMetadataName(typeof(CancellationToken).FullName)))
+                {
+                    insertionIndex++;
+                }
+            }
+
+            return insertionIndex;
+        }
+
+        /// <summary>
+        /// Goes through all of the invocations and replaces the expression with identifiers from the invocation 
+        /// arguments and rewrites the call site with the updated expression as a new argument
+        /// </summary>
+        private async Task<Solution> RewriteCallSitesWithNewMethodAsync(SemanticDocument semanticDocument,
+                                                                               TExpressionSyntax expression,
+                                                                               IMethodSymbol methodSymbol,
+                                                                               bool allOccurrences,
+                                                                               string parameterName,
+                                                                               ImmutableDictionary<Document, List<TInvocationExpressionSyntax>> callSites,
+                                                                               bool trampoline,
+                                                                               CancellationToken cancellationToken)
+        {
+            var firstCallSite = callSites.Keys.First();
+            var currentSolution = firstCallSite.Project.Solution;
+            var modifiedSolution = currentSolution;
+            var newMethodIdentifier = methodSymbol.Name + "_" + parameterName;
+
+            foreach (var grouping in callSites.GroupBy(kvp => kvp.Key.Project))
+            {
+                var project = grouping.Key;
+                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                foreach (var keyValuePair in grouping)
+                {
+                    var document = keyValuePair.Key;
+                    var invocationExpressionList = keyValuePair.Value;
+                    var generator = SyntaxGenerator.GetGenerator(document);
+                    var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
+                    var invocationSemanticModel = compilation.GetSemanticModel(await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false));
+
+                    var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var editor = new SyntaxEditor(root, generator);
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                    var oldMethodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsMethodDeclaration(node), ascendOutOfTrivia: true)!;
```

Comment:
this is very confusing.  we're mixing and matchign between two different documents (potentially in different projects) without clear namign about which is being referenced.

Reply:
Will message you about this